slack_sdk/socket_mode/builtin/internals.py

```diff
@@ -59,6 +59,11 @@ def _establish_new_socket_connection(
         sock.settimeout(receive_timeout)
         sock.connect(proxy_addr[4])  # proxy address
         message = [f"CONNECT {server_hostname}:{server_port} HTTP/1.0"]
+        if parsed_proxy.username is not None and parsed_proxy.password is not None:
+            # In the case where the proxy is "http://{username}:{password}@{hostname}:{port}"
+            raw_value = f"{parsed_proxy.username}:{parsed_proxy.password}"
+            auth = b64encode(raw_value.encode("utf-8")).decode("ascii")
+            message.append(f"Proxy-Authorization: Basic {auth}")
```

Comment:
If a developer needs a different header for authentication (e.g., digest auth), we recommend using `proxy_headers` instead.